pybop/parameters/parameter.py

```diff
@@ -143,3 +146,245 @@ def set_bounds(self, bounds=None):
+
+        self.update_bounds()
+
+    def remove_parameter(self, parameter_name):
+        """
+        Construct the parameter class with a name, initial value, prior, and bounds.
+        """
+        if not isinstance(parameter_name, str):
+            raise TypeError("The input parameter_name is not a string.")
+        if parameter_name not in self.param.keys():
+            raise ValueError("This parameter does not exist in the Parameters object.")
+
+        # Remove the parameter
+        self.param.pop(parameter_name)
+
+        self.update_bounds()
+
+    def update_bounds(self):
+        """
+        Set bounds, for either all or no parameters.
+        """
+        all_unbounded = True  # assumption
+        self.bounds = {"lower": [], "upper": []}
+
+        for param in self.param.values():
+            if param.bounds is not None:
+                self.bounds["lower"].append(param.bounds[0])
+                self.bounds["upper"].append(param.bounds[1])
+                all_unbounded = False
+            else:
+                self.bounds["lower"].append(-np.inf)
+                self.bounds["upper"].append(np.inf)
+        if all_unbounded:
+            self.bounds = None
+
+        return self.bounds
+
+    def update(self, values):
+        """
+        Set value of each parameter.
+        """
+        for i, param in enumerate(self.param.values()):
+            param.update(value=values[i])
+
+    def rvs(self, n_samples):
+        """
+        Draw random samples from each parameter's prior distribution.
+
+        The samples are constrained to be within the parameter's bounds, excluding
+        a predefined margin at the boundaries.
+
+        Parameters
+        ----------
+        n_samples : int
+            The number of samples to draw.
+
+        Returns
+        -------
+        array-like
+            An array of samples drawn from the prior distribution within each parameter's bounds.
+        """
+        all_samples = np.zeros(len(self))
+
+        for i, param in enumerate(self.param.values()):
+            samples = param.rvs(n_samples)
+
+            # Constrain samples to be within bounds
+            if param.bounds is not None:
+                offset = param.margin * (param.upper_bound - param.lower_bound)
+                samples = np.clip(
+                    samples, param.lower_bound + offset, param.upper_bound - offset
+                )
+
+            all_samples[i] = samples
+
+        return all_samples
+
+    def get_sigma0(self):
+        """
+        Set initial standard deviation, for either all or no parameters.
+        """
+        all_have_sigma = True  # assumption
+        sigma0 = []
+
+        for param in self.param.values():
+            if hasattr(param.prior, "sigma"):
+                sigma0.append(param.prior.sigma)
+            else:
+                all_have_sigma = False
+        if not all_have_sigma:
```

Comment:
I wonder if a better implementation would be to insert either `Nan` or `np.inf` and return the list. This would ensure that we get a value of`sigma0` for all defined parameters.

Reply:
Good idea, let's implement and test this in another PR